rllib/core/optim/tests/test_rl_optimizer_tf.py

```diff
@@ -0,0 +1,477 @@
+            self.output_dim = config.hidden_layers[-1]
+        else:
+            self.output_dim = config.output_dim
+
+        self._layers = tf.keras.Sequential(self._layers)
+
+        self._input_specs = self.input_specs()
+
+    def input_specs(self):
+        return TFTensorSpecs("b, h", h=self.input_dim)
+
+    @check_specs(input_spec="_input_specs")
+    def call(self, x, training=False):
+        return self._layers(x)
+
+
+class BCTFModule(TFRLModule):
+    def __init__(self, config: FCConfig) -> None:
+        super().__init__(config)
+        self.policy = TFFCNet(config)
+
+    def input_specs(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TFTensorSpecs("b, do", do=obs_dim),
+                "actions": TFTensorSpecs("b"),
+            }
+        )
+
+    def output_specs(self) -> ModelSpec:
+        return ModelSpec({"action_dist": tfp.distributions.Distribution})
+
+    def input_specs_exploration(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TFTensorSpecs("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_inference(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TFTensorSpecs("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_train(self) -> ModelSpec:
+        return self.input_specs()
+
+    def output_specs_exploration(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_inference(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_train(self) -> ModelSpec:
+        return self.output_specs()
+
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_logits_inference = tf.argmax(action_logits, axis=-1)
+        action_dist = tfp.distributions.Deterministic(action_logits_inference)
+        return {"action_dist": action_dist}
+
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        return self._forward_inference(batch)
+
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+
+        action_dist = tfp.distributions.Categorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    def call(self, batch, training=False):
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+
+        action_dist = tfp.distributions.Categorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    def is_distributed(self) -> bool:
+        return False
+
+    def make_distributed(self, dist_config: Mapping[str, Any] = None) -> None:
+        pass
+
+    def get_state(self) -> Mapping[str, Any]:
```

Comment:
You don't need to override this, right? self.get_state() already has a good enough default implementation.